server/lib/routes/otp.js

```diff
@@ -76,12 +85,25 @@ async function otpRoutes(server) {
           }
         )
 
-        push.send({
+        const { name, version } = body.packageInfo
+
+        const notification = {
           subscription: subscription.data(),
           secretId,
           uniqueId,
           token
-        })
+        }
+
+        const hasPackageInfo = !!name || !!version
+
+        if (hasPackageInfo) {
+          notification.packageInfo = {
+            ...(version && { version }),
+            ...(name && { name })
+          }
+        }
```

Comment:
can we write this logic a little better? I'm confident that there's no need to check each property individually. as long as the body exists and has a packageInfo property, then we should just send whatever's in there

Reply:
Yeah I think that makes sense.
Should we validate the name and version and return an error for the caller if they are not present? mostly in case there's a typo or something else like that which could be a little annoying to debug.

Reply:
Just updated the code. It did require a loop through the HTTP methods so I could add the schema conditionally as it would fail trying to validate the body of the GET request. The loop does cause the diff to be a little weird again but I think it's still better. My other option would be just repeating the declaration (seeing that this would eventually be removed anyway) but then the code coverage dropped and I would have to repeat tests for each method so I think the loop should work fine.